networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/rpc/RpcHandler.java

```diff
@@ -157,10 +174,18 @@ public RpcStream getRpcStream() {
 
     @Override
     protected void channelRead0(final ChannelHandlerContext ctx, final ByteBuf msg) {
-      if (rpcRequestHandler != null) {
-        rpcRequestHandler.onData(nodeId, rpcStream, msg);
-      } else {
-        bufferedData.add(msg);
+      if (outputStreamClosed) {
+        // Discard any data if output stream has been closed
+        return;
+      }
+      try {
+        // TODO - we may want to optimize this to pass on ByteBuf's directly and manage their
+        //  garbage collection rather than immediately copying these bytes
+        final Bytes bytes = Bytes.wrapByteBuf(msg);
```

Comment:
On the plus side, copying the data means we don't have to manage retaining and releasing the `ByteBuf` if we jump threads later.  So unless this shows up as a significant performance issue it would be great to leave as-is.